Source/DafnyCore/AST/DafnyAst.cs

```diff
@@ -162,6 +162,15 @@ public interface IAttributeBearingDeclaration {
     Attributes Attributes { get; }
   }
 
+  static class AttributeBearingDeclaration {
+
+    public static bool IsExtern(this IAttributeBearingDeclaration me) =>
+      DafnyOptions.O.AllowExterns && Attributes.Contains(me.Attributes, "extern");  // TODO do earlier check to complain about any usages of :extern if allow externs is false.
```

Comment:
Would it be feasible to include this change in this PR, or is it a lot of extra work? If it's the latter, could we have an issue for it?

Reply:
I've removed the TODO since I'm sure whether this option is used. It's not part of the new CLI